### ISSUE TITLE
provide new property for sequential work of "CompositeFuture.all"

### ISSUE THREAD

**Issue by nimo23** [eclipse/vert.x#2018]
Please provide a optional property for "CompositeFuture.all" which makes the Composite of a list to run sequential (from first index to last index of the list) or in parallel (default).

For example:

```
CompositeFuture.all(listOfFutures, Composite.SEQUENTIAL)
.setHandler(r->{
			log.debug("step 1 completed: {}", step1.isComplete());
			log.debug("step 2 completed: {}", step2.isComplete());
	    });;

```
This eliminates the burden to use chaining compose with callback hell..

**Comment by vietj** [comment]
it should rather be a `sequence` method API than a mode.

**Comment by vietj** [comment]
please also define precisely what you mean by sequential

**Comment by nimo23** [comment]
sequential means: make "step 2" after "step 1" terminates.

Actually, I have to use http://vertx.io/docs/vertx-core/java/#_sequential_composition, to make somethink like this:

```
// sequence code without vertx
// at first step 1
step1();
// after termination of step1, proceed with step2
step2();

```

**Comment by vietj** [comment]
and there is no relationship between step1 and step2 beside the timing

**Comment by nimo23** [comment]
there is a relationship between these two steps:

In each step, I make GET-Requests and the GET-Requests of step2() depends of the GET-Requests of step1().

**Comment by nimo23** [comment]
It would be nice to have somethink like this:

private Future<Void> runStep1(){}
private Future<Void> runStep2(){}

CompositeFuture.all({runStep1, runStep2}, Composite.SEQUENTIAL)

**Comment by vietj** [comment]
I think this is out of scope of the CompositeFuture

**Comment by nimo23** [comment]
I am thinking of "composing" a series of method calls by different Compositon-Patterns (Sequential).

**Comment by nimo23** [comment]
However, is there a better alternative as this:

```
@Override
public void start(Future<Void> fut) throws IOException {

Future<Void> step_1 = Future.future();
    step_1.compose(res -> {
        // If the future succeeded
        Future<Void> step_2 = step_1();

        step_2.compose(res2 -> {
            step_2();

        }, Future.future().setHandler(handler -> {
            // If the future failed
        }));

        //I dont need that
    }, Future.future().setHandler(handler -> {
        // If the future failed
    }));

}

public void step_1(){
..
}

public void step_2(){
..
}
```

I dont need handlers (for errors) but I am forced by the api. And nested callback..

**Comment by nimo23** [comment]
Maybe somethink like 

`ComposeTransactions(compose({future1, future2), parallel), compose({future3,future5}, parallel), sequential);`

**Comment by nimo23** [comment]
Ok, I have found that kind within rxjava. So no need for that. However, if you think this would better vertx sync handling, then feel free, to reopen this issue.

**Comment by vietj** [comment]
I believe it's better to be addressed with RxJava
